### PR TITLE
[Utils] Node homophily measure

### PR DESCRIPTION
**Pull request by mufeili**
## Description
I'll implement the rest two measures once everyone is comfortable with the practice in this PR.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [x] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/homophily.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
+"""Utils for tacking graph homophily and heterophily"""
+from . import backend as F, function as fn
+
+__all__ = ["node_homophily"]
+
+
+def node_homophily(graph, y):
+    r"""Homophily measure from `Geom-GCN: Geometric Graph Convolutional Networks
```
Comment (frozenbugs) [inline python/dgl/homophily.py:8]: measurement

File: python/dgl/homophily.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    <https://arxiv.org/abs/2002.05287>`__
+
+    We follow the practice of a later paper `Large Scale Learning on
```
Comment (frozenbugs) [inline python/dgl/homophily.py:11]: avoid using `later`, `new` in code, the code will be here for long, but the paper will not be new any more :)

File: tests/python/common/test_homophily.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import backend as F
+
+import dgl
+from test_utils import parametrize_idtype
+
+
+@unittest.skipIf(dgl.backend.backend_name == "tensorflow", reason="Skip TF")
+@parametrize_idtype
+def test_node_homophily(idtype):
+    # NOTE: If you want to update this test case, remember to update the
```
Comment (frozenbugs) [inline tests/python/common/test_homophily.py:12]: If we want to introduce this practice, let's follow this format:

```
# IfChangeThenChange: python/dgl/homophily.py
# Update the docstring example.
```

Comment (frozenbugs) [reply tests/python/common/test_homophily.py:12]: Added in the conventions:
https://docs.google.com/document/d/1NGXPg4U67ZpMDVxxkb-SprEx11d8zqNB_0cLweTL90Y/edit#

**Review by frozenbugs — APPROVED** [review]

File: python/dgl/homophily.py (hunk 1 above)
Comment (mufeili) [reply python/dgl/homophily.py:8]: Measure is the term used in this kind of papers: https://arxiv.org/abs/2110.14446

File: python/dgl/homophily.py (hunk 2 above)
Comment (mufeili) [reply python/dgl/homophily.py:11]: It emphasizes the fact that the term was not introduced in the original paper.

File: tests/python/common/test_homophily.py (hunk 3 above)
Comment (mufeili) [reply tests/python/common/test_homophily.py]: updated

**Pull request merged**